lgc/builder/ImageBuilder.cpp

```diff
@@ -1331,6 +1337,15 @@ Value *ImageBuilder::CreateImageQuerySize(unsigned dim, unsigned flags, Value *i
 
   unsigned sizeComponentCount = getImageQuerySizeComponentCount(dim);
 
+  if (!(flags & ImageFlagNonUniformImage)) {
+    for (unsigned i = 0; i < sizeComponentCount; i++) {
+      Value *index = getInt32(i);
+      Value *element = CreateExtractElement(intResInfo, index, instName);
+      element = CreateIntrinsic(Intrinsic::amdgcn_readfirstlane, {}, element);
+      intResInfo = CreateInsertElement(intResInfo, element, index);
+    }
+  }
+
```

Comment:
I don't think this is correct: `lod` could be a non-uniform value, and in that case the result would also be non-uniform.

Reply:
Yes, you are right. I will try to check it.

Reply:
@nhaehnle, do you think if we still need to go ahead on this pull request? Or just move to implement #845.

Reply:
If #845 is done we don't need this anymore.